src/mapContainer.ts

```diff
@@ -645,13 +713,181 @@ export class MapContainer {
+        for (let [k, fwm] of fileMarkerMap) {
+            this.generateEdgesFromFileWithMarkers(
+                fwm,
+                fileMarkerMap,
+                edges,
+                nodesSeen
+            );
+        }
+        return edges;
+    }
+
+    private generateEdgesFromFileWithMarkers(
+        source: FileWithMarkers,
+        fileMarkerMap: Map<string, FileWithMarkers>,
+        edges: Map<string, Edge>,
+        nodesSeen: Set<string>
+    ) {
+        const file = source.file;
+        const path = file.path;
+        if (nodesSeen.has(path)) {
+            return;
+        }
+        nodesSeen.add(path);
+        const fileCache = this.app.metadataCache.getFileCache(file);
+        for (let link of fileCache?.links || []) {
+            let destination = this.app.metadataCache.getFirstLinkpathDest(
+                link.link,
+                path
+            );
+            if (destination?.path && fileMarkerMap.has(destination.path)) {
+                // both the source file and the destination file have markers;
+                // let's connect them and create edges
+                let destinationFileWithMarkers = fileMarkerMap.get(
+                    destination.path
+                );
+                for (let sourceMarker of source.markers) {
+                    for (let destinationMarker of destinationFileWithMarkers.markers) {
+                        let edge: Edge = {
+                            v1: new leaflet.LatLng(
+                                sourceMarker.location.lat,
+                                sourceMarker.location.lng
+                            ),
+                            v2: new leaflet.LatLng(
+                                destinationMarker.location.lat,
+                                destinationMarker.location.lng
+                            ),
+                        };
+                        edges.set(
+                            `${edge.v1.toString()}<<->>${edge.v2.toString()}`,
+                            edge
+                        );
+                    }
+                }
+                // continue to traverse files and links recursively
+                this.generateEdgesFromFileWithMarkers(
+                    destinationFileWithMarkers,
+                    fileMarkerMap,
+                    edges,
+                    nodesSeen
+                );
+            }
+        }
     }
 
     private newLeafletMarker(marker: FileMarker): leaflet.Marker {
+        let icon:
+            | leaflet.Icon<leaflet.ExtraMarkers.IconOptions>
+            | leaflet.DivIcon;
+        if (marker.icon) {
+            icon = marker.icon;
+        } else {
+            icon = new leaflet.Icon.Default();
+        }
+
         let newMarker = leaflet.marker(marker.location, {
-            icon: marker.icon || new leaflet.Icon.Default(),
+            icon: icon,
+            draggable: this.settings.drawEdgesBetweenMarkers,
```

Comment:
This doesn't seem to be related to whether edges are active and needs to be a separate setting. But much more importantly, it might require some locking mechanism, as accidentally dragging a marker and irreversibly destroying a location you logged seems way too easy. How about having an "unlock position" popup menu item for a marker, which would make it draggable?

Reply:
I will make both of these changes. I fully agree.